pkg/fuzzy/google_translate.go

```diff
@@ -119,43 +118,60 @@ func (g *GoogleTranslate) Translate(
 		return &model.Messages{Language: messages.Language, Original: messages.Original}, nil
 	}
 
-	n := len(messages.Messages)
 	translatedMessages := model.Messages{
 		Language: targetLanguage,
 		Original: messages.Original,
-		Messages: make([]model.Message, 0, n),
+		Messages: make([]model.Message, 0, len(messages.Messages)),
 	}
 
-	for low := 0; low < n; low += googleTranslateRequestlimit {
-		high := low + googleTranslateRequestlimit
+	// Split text from messages into batches to avoid exceeding
+	// googleTranslateRequestLimit or googleTranslateCodePointsLimit.
+
+	var codePointsInBatch, msgsInBatch int
+
+	batch := make([]string, 0)
```

Comment:
What is the benefit of a zero-length empty slice over a nil slice? Can we predict the required size?

Reply:
Lint was complaining about pre-alloc. We could pre-allocate capacity using translateRequestLimit, but mostly I think batch size will be less, about 600.

Reply:
We know, for sure, the minimum is at least one, and also the maximum size is known - googleTranslateRequestLimit. We can reserve the upper limit, we may not use all reserved memory but the size is quite small.

```suggestion
	batch := make([]string, 0, googleTranslateRequestLimit)
```